app/src/main/java/com/teamagam/gimelgimel/app/map/esri/MapDragEventsEmitterTouchListenerDecorator.java

```diff
@@ -0,0 +1,112 @@
+package com.teamagam.gimelgimel.app.map.esri;
+
+import android.view.MotionEvent;
+import android.view.View;
+import com.esri.android.map.MapOnTouchListener;
+import com.esri.android.map.MapView;
+import com.teamagam.gimelgimel.app.map.MapDragEvent;
+import com.teamagam.gimelgimel.domain.map.entities.geometries.PointGeometry;
+import io.reactivex.functions.BiFunction;
+import io.reactivex.subjects.Subject;
+
+class MapDragEventsEmitterTouchListenerDecorator implements View.OnTouchListener {
```

Comment:
As a decorator I would have expected it to extend `MapOnTouchListener` and not `View.OnTouchListener`. It will also allow u to remove all those dummy implementations of the many methods the interface enforces.